### PR TITLE
Skia sharp drawing lib

### PR DESCRIPTION
**Pull request by barnhill**
@binki  This will remove the dependency on System.Drawing.Common which isnt cross platform compatible.  To replace it this will transition to utilizing SkiaSharp as the drawing lib for cross platform support.

UPCE and EAN8 barcodes are broken but I checked prior to this and they were broken prior to this and generate the same barcode so I will consider it a separate issue.

This will also introduce a lot of tests for various symbologies.  However the coverage is incomplete but adds more than we have today.

### CODE REVIEW

**Comment by tlogik** [comment]
Hi @binki  and @barnhill 
What is required for the PR to go thru in relation to review?

**Comment by tlogik** [comment]
> I could merge it today but wouldn't mind some eyes on it to make sure I didn't break something. I think labels are kinda messed up but those need either removed or fixed as they don't work right even before this change

I there something i could do here?
I do not have the overview of the code as you do, i just found it, but i might be able to do some assistance but for me to review you change it would require me to really dive into the code and that will take some time no matter what.

Also this is a breaking change right Will it just be a Major version update?
Lastly it seems the documentation might need to be updated as well (I could update the PR with an updated Readme)

Let me know if i can assist :-)

**Comment by ozymandias13** [comment]
I'm very interested in this update as well, so I'll give it a review and make sure nothing jumps out at me.

File: BarcodeStandard/BarcodeLib.cs [hunk 1]
```diff
@@ -1172,156 +893,152 @@ public class Utf8StringWriter : StringWriter
...
-                Encoded_Value = null;
-                _Country_Assigning_Manufacturer_Code = null;
-                _ImageFormat = null;
-            }
+            RawData = null;
+            EncodedValue = null;
+            CountryAssigningManufacturerCode = null;
         }
-
-        // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
```
Comment (ozymandias13) [inline BarcodeStandard/BarcodeLib.cs:1320]: Did you mean to uncomment this is you haven't done the TODO on what was line 1300 (now line 1020)?

**Review by ozymandias13 — APPROVED** [review]
TL;DR - LGTM

One minor question. Otherwise, the changes are all consistent on how you integrated the Skia stuff and I was able to resolve major differences, with a cursory understanding. So, it all seems to make sense to me and I like the test coverage you added.

File: BarcodeStandard/BarcodeLib.cs (hunk 1 above)
Comment (ozymandias13) [reply BarcodeStandard/BarcodeLib.cs:1320]: Nvm. I see the other issue referenced and why this was put back in. Looks good!

**Comment by barnhill** [comment]
The big changes were around migrating to SkiaSharp to do the drawing but there were some issues with labels (which are totally screwed up) but it looks like they pre-dated this update.  Im not a huge fan of how the labels were written originally (by me).  Also the UPC-E barcode seems to not be encoding properly to read but I tested that and it pre-dated this as well.  I will take a look at that separately.  Everything else seems to be ok.

**Comment by barnhill** [comment]
> > I could merge it today but wouldn't mind some eyes on it to make sure I didn't break something. I think labels are kinda messed up but those need either removed or fixed as they don't work right even before this change
> 
> I there something i could do here? I do not have the overview of the code as you do, i just found it, but i might be able to do some assistance but for me to review you change it would require me to really dive into the code and that will take some time no matter what.
> 
> Also this is a breaking change right Will it just be a Major version update? Lastly it seems the documentation might need to be updated as well (I could update the PR with an updated Readme)
> 
> Let me know if i can assist :-)

I think this large of a change warrants a major version number update. :)

**Comment by barnhill** [comment]
Also added builds on windows and linux and .net 6 and 7 for verification :)

**Comment by barnhill** [comment]
I still see a slight deviation from center on the generic label.  Need to fix this before merging.

**Comment by barnhill** [comment]
Ok I think I have the centering of the generic symbology labels fixed, also fixed the EAN-13 labels.  Anyone wanna give this a whirl to see if its ready to go?

**Review by ozymandias13 — APPROVED** [review]
I'm approving a second time after the last round of changes. I also pulled it down and did some spot checking. My domain knowledge here is limited, so I checked on what I know, which isn't super comprehensive. However, the projects did build without any intervention after pulling down this branch. All the tests passed and the sample app runs just fine.

**Comment by barnhill** [comment]
Im going to do some more spot checking and if all still looks good Im going to deploy it.  It does contain some breaking changes with the API but thats why its a major version bump too.

**Comment by ozymandias13** [comment]
> Im going to do some more spot checking and if all still looks good Im going to deploy it. It does contain some breaking changes with the API but thats why its a major version bump too.

For what it's worth, as another date point: we pulled this branch down and built the lib and pulled it into our project and updated our usage to test the new API for the standard 2 of 5 and it worked just fine.

**Comment by barnhill** [comment]
Just went through and generated a bunch of different types and scanned them in 2 online barcode scanners and it detected them all correctly.  I think we are good to go.
